crates/driver/src/tests/cases/protocol_fees.rs

```diff
@@ -111,6 +111,320 @@ async fn protocol_fee_test_case(test_case: TestCase) {
+    let fee_policy_surplus = Policy::Surplus {
+        factor: 0.5,
+        // high enough so we don't get capped by volume fee
+        max_volume_factor: 1.0,
+    };
+    let fee_policy_volume = Policy::Volume { factor: 0.25 };
+    let test_case = TestCase {
+        fee_policy: vec![fee_policy_volume, fee_policy_surplus],
+        order: Order {
+            sell_amount: 60.ether().into_wei(),
+            buy_amount: 40.ether().into_wei(),
+            side: order::Side::Buy,
+        },
+        execution: Execution {
+            // -> First fee policy:
+            // 25% of the solver proposed sell volume is kept by the protocol
+            // solver executes at the adjusted limit price ( 50 / (1 + 0.25) = 40 )
+            // Fee = 50 ETH (limit price) - 40 ETH => 10 ETH
+            // -> Second fee policy:
+            // New buy amount in the Order: 40 ETH + 10 ETH (fee) = 50 ETH
+            // New surplus: 10 ETH, fee 5 ETH
+            // Total fee: 15 ETH
+            solver: Amounts {
+                sell: 40.ether().into_wei(),
+                buy: 40.ether().into_wei(),
+            },
+            // driver executes at limit price
+            driver: Amounts {
+                sell: 55.ether().into_wei(),
+                buy: 40.ether().into_wei(),
+            },
+        },
+        expected_score: 20.ether().into_wei(),
+        fee_handler: FeeHandler::Driver,
+    };
+    protocol_fee_test_case(test_case).await;
+}
+
+#[tokio::test]
+#[ignore]
+async fn surplus_and_price_improvement_protocol_fee_sell_order_not_capped() {
+    let fee_policy_surplus = Policy::Surplus {
+        factor: 0.5,
+        // high enough so we don't get capped by volume fee
+        max_volume_factor: 0.9,
+    };
+    let fee_policy_price_improvement = Policy::PriceImprovement {
+        factor: 0.5,
+        // high enough so we don't get capped by volume fee
+        max_volume_factor: 0.9,
+        quote: Quote {
+            sell: 49.ether().into_wei(),
+            buy: 40.ether().into_wei(),
+            network_fee: 1.ether().into_wei(),
+        },
+    };
+    let test_case = TestCase {
+        fee_policy: vec![fee_policy_price_improvement, fee_policy_surplus],
+        order: Order {
+            sell_amount: 50.ether().into_wei(),
+            // Demanding to receive more than quoted (out-market)
+            buy_amount: 50.ether().into_wei(),
+            side: order::Side::Sell,
+        },
+        execution: Execution {
+            // -> First fee policy:
+            // Receive 10 ETH more than quoted, half of which gets captured by the protocol (10 ETH)
+            // Fee = 10 ETH * 0.5 => 5 ETH
+            // -> Second fee policy:
+            // New buy amount in the Order: 50 ETH + 5 ETH (fee) = 55 ETH
+            // New surplus: 5 ETH, fee 2.5 ETH
+            // Total fee: 7.5 ETH
+            solver: Amounts {
+                sell: 20.ether().into_wei(),
+                buy: 30.ether().into_wei(),
```

Comment:
Is this a test for a partially fillable order? Otherwise I'm not sure how we arrive at these values.
If it is indeed a partially fillable test it would be good to mention it somewhere. (same goes for all partially fillable tests)

Reply:
@MartinquaXD of course! I changed the numbering and the wording to make it clearer. Please, let me know if it is ok like this :pray: